babel.config.js

```diff
@@ -3,6 +3,7 @@ module.exports = {
     `@babel/plugin-transform-modules-commonjs`,
     `@babel/plugin-proposal-optional-chaining`,
     `@babel/plugin-proposal-nullish-coalescing-operator`,
+    [`@babel/plugin-transform-typescript`, {allowDeclareFields: true}],
```

Comment:
Should be set as an option on `@babel/preset-typescript` instead

Reply:
I don't think it's possible to make it work, unfortunately. When using TS declare fields, the TS transform plugin has to be loaded before the class fields plugin, but presets are loaded after plugins, so we have to use it as a plugin unless I'm missing something. Also, when I tried removing the preset and only keeping the plugin, it complained about not supporting JSX, so... I kept it.